services/ui-src/src/utils/other/mask.tsx

```
@@ -1,104 +1,88 @@
  */
-export function convertToCommaSeparatedRatioString(value: string): string {
-  // Remove all characters except digits and decimal points.
-  value = value.replace(/[^\d.:]/g, "");
+export function convertToThousandsSeparatedRatioString(value: string): {
+  maskedValue: string;
+  cleanedValue: string;
+} {
+  // Clean value
+  const cleanedInput = cleanRatioInput(value);
+  if (!cleanedInput.isValid) return { maskedValue: value, cleanedValue: value };
 
   // Grab the left and right side of the ratio sign
-  const values = value.split(":");
-
-  // Begin creating the final output
-  let cleanedValue = "";
+  const values = cleanedInput.cleanedValue.split(":");
 
   // Create the left side of the output and make the number (if provided) pretty
-  if (values[0] != "") cleanedValue += convertToCommaSeparatedString(values[0]);
-  else cleanedValue += "";
-
-  // Put in the ratio sign in the middle of the two numbers
-  cleanedValue += ":";
+  values[0] = convertToThousandsSeparatedString(values[0]).maskedValue;
 
   // Create the right side of the output and make the number (if provided) pretty
-  if (values.length >= 2 && values[1] != "")
-    cleanedValue += convertToCommaSeparatedString(values[1]);
-  else cleanedValue += "";
+  values[1] = convertToThousandsSeparatedString(values[1]).maskedValue;
 
-  return cleanedValue;
+  const maskedValue = values.join(":");
+  return { maskedValue: maskedValue, cleanedValue: cleanedInput.cleanedValue };
 }
 
-/**
- * Converts string to the appropriate custom masked format
- * @param {String} value
- * @returns {String}
- */
-export function maskValue(
+// if valid custom mask, return masked value; else return value
+export const applyMask = (
   value: string,
-  mask: keyof typeof customMaskMap
-): string {
-  const selectedCustomMask = customMaskMap[mask];
-  if (isValidNumericalString(value)) {
-    return selectedCustomMask(value);
-  } else return value;
-}
+  maskName?: keyof typeof maskMap | null
```

What happens if `maskName` is undefined here?

---

it defaults to the thousand separated mask!